internal/pkg/vm/actor/builtin/initactor/init.go

```diff
@@ -99,3 +138,190 @@ func (*Impl) GetNetwork(ctx runtime.Runtime) (string, uint8, error) {
 
 	return state.Network, 0, nil
 }
+
+// GetActorIDForAddress looks up the actor id for a filecoin address.
+func (a *Impl) GetActorIDForAddress(rt runtime.Runtime, addr address.Address) (types.Uint64, uint8, error) {
+	if err := rt.Charge(actor.DefaultGasCost); err != nil {
+		return 0, internal.ErrInsufficientGas, errors.RevertErrorWrap(err, "Insufficient gas")
+	}
+
+	var state State
+	err := actor.ReadState(rt, &state)
+	if err != nil {
+		return 0, errors.CodeError(err), err
+	}
+
+	ctx := context.TODO()
```

Comment:
A thought @icorderi : would be handy for the Runtime to provide a context, since numerous Go calls will need it (as a Go-specific thing). This also gives some hook to terminating infinite loops and the like.